### PR TITLE
bug fixes and support for ccr mpi

### LINKED ISSUE #8: Produce documentation/code changes to build/run on CCR

### PR DESCRIPTION
**Pull request by mtmcgurn-buffalo**
this pr should close #8 with documentation on the [wiki](https://github.com/UBCHREST/framework/wiki/Developing-on-CCR)

### CODE REVIEW

File: src/mesh.c [hunk 1]
```diff
@@ -10,5 +10,20 @@ PetscErrorCode CreateMesh(MPI_Comm comm, DM *dm, PetscBool simplex, PetscInt dim
 
     ierr = DMViewFromOptions(*dm, NULL, "-dm_view");CHKERRQ(ierr);
 
+    // distribute the mesh
```
Comment (knepley) [inline src/mesh.c:13]: This is fine, and matches prior examples. However, I am moving to an interface where distribution is handled automatically in DMSetFromOptions() using the -dm_distribute option. This means less code in the application to maintain. We can always do this later, but I thought I would mention it.

**Review by knepley — APPROVED** [review]
I left one comment

File: src/mesh.c (hunk 1 above)
Comment (mtmcgurn-buffalo) [reply src/mesh.c]: Thanks!  I removed the manual distribute and added the option to the test.

**Comment by mtmcgurn-buffalo** [comment]
@knepley Would you mind doing another review with the -dm_distribute changes when you get a chance please?

File: tests/lowMachNumberTests.cpp [hunk 2]
```diff
@@ -588,6 +588,23 @@ INSTANTIATE_TEST_SUITE_P(
...
+        (LowMachMMSParameters){
+            .mpiTestParameter = {.nproc = 4,
+                                 .expectedOutputFile = "outputs/2d_tri_p2_p1_p1_nproc4",
+                                 .arguments = "-dm_plex_separate_marker -dm_refine 2 "
+                                              "-vel_petscspace_degree 2 -pres_petscspace_degree 1 -temp_petscspace_degree 1 "
+                                              "-dmts_check .001 -ts_max_steps 4 -ts_dt 0.1 "
+                                              "-ksp_type fgmres -ksp_gmres_restart 10 -ksp_rtol 1.0e-9 -ksp_error_if_not_converged "
+                                              "-pc_type fieldsplit -pc_fieldsplit_0_fields 0,2 -pc_fieldsplit_1_fields 1 -pc_fieldsplit_type schur -pc_fieldsplit_schur_factorization_type full "
+                                              "-fieldsplit_0_pc_type lu "
+                                              "-fieldsplit_pressure_ksp_rtol 1e-10 -fieldsplit_pressure_pc_type jacobi -dm_distribute"},
```
Comment (knepley) [inline tests/lowMachNumberTests.cpp:600]: I would put this up on line 594 with the other DM options

**Review by knepley — DISMISSED** [review]
I have one suggested change

File: tests/lowMachNumberTests.cpp (hunk 2 above)
Comment (mtmcgurn-buffalo) [reply tests/lowMachNumberTests.cpp]: @knepley Thanks!  I've cleaned up the dm_distribute and reduced the refine level to speed up the tests

**Pull request merged**